File: fincapp-backend/src/FincApp.API/DTOs/SyncWeightLogDto.cs
using System;
using System.Text.Json.Serialization;

namespace FincApp.API.DTOs;

public class SyncWeightLogDto
{
    [JsonPropertyName("id")]
    public Guid? Id { get; set; }

    [JsonPropertyName("animal_id")]
    public Guid AnimalId { get; set; }

    [JsonPropertyName("weight_kg")]
    public decimal WeightKg { get; set; }

    [JsonPropertyName("log_date")]
    public DateTime LogDate { get; set; }
}

File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/network/dto/SyncWeightLogDto.java
package com.irwi.fincapp.network.dto;

import com.google.gson.annotations.SerializedName;

public class SyncWeightLogDto {
    public String id;
    @SerializedName("animal_id") public String animalId;
    @SerializedName("weight_kg") public double weightKg;
    @SerializedName("log_date") public String logDate;
}
